x/permission/types/types.go

```diff
@@ -211,30 +214,48 @@ func (s *Statement) ValidateBasic(resType resource.ResourceType) error {
 	return nil
 }
 
-func (s *Statement) ValidateAfterNagqu(resType resource.ResourceType) error {
-	if s.Effect == EFFECT_UNSPECIFIED {
-		return ErrInvalidStatement.Wrap("Please specify the Effect explicitly. Not allowed set EFFECT_UNSPECIFIED")
-	}
-	switch resType {
-	case resource.RESOURCE_TYPE_UNSPECIFIED:
-		return ErrInvalidStatement.Wrap("Please specify the ResourceType explicitly. Not allowed set RESOURCE_TYPE_UNSPECIFIED")
-	case resource.RESOURCE_TYPE_BUCKET:
-		for _, r := range s.Resources {
-			_, err := regexp.Compile(r)
-			if err != nil {
-				return ErrInvalidStatement.Wrapf("The Resources regexp compile failed, err: %s", err)
+func (s *Statement) ValidateRuntime(ctx sdk.Context, resType resource.ResourceType) error {
+	if ctx.IsUpgraded(upgradetypes.Nagqu) {
```

Comment:
we delete `if s.Effect == EFFECT_UNSPECIFIED`, will it cause consensus issue?

Reply:
`validateBasic` has covered this, i think.